File: Python/Program/Vehicle.py
class Vehicle:
    __NoPlat = ""
    __Merk = ""
    __TahunProduksi = 0
    __Warna = ""
        
    def __init__(self, NoPlat, Merk, TahunProduksi, Warna):
        self.__NoPlat = NoPlat
        self.__Merk = Merk
        self.__TahunProduksi = TahunProduksi
        self.__Warna = Warna
    
    # setter getter
    def setNoPlat(self, NoPlat):
        self.__NoPlat = NoPlat
        
    def getNoPlat(self):
        return self.__NoPlat
    
    def setMerk(self, Merk):
        self.__Merk = Merk
        
    def getMerk(self):
        return self.__Merk
    
    def setTahunProduksi(self, TahunProduksi):
        self.__TahunProduksi = TahunProduksi
        
    def getTahunProduksi(self):
        return self.__TahunProduksi
    
    def setWarna(self, Warna):
        self.__Warna = Warna
        
    def getWarna(self):
        return self.__Warna
    
    

File: CPP/Program/Vehicle.cpp
#pragma once

#include <string>
#include <iostream>

using namespace std;

class Vehicle {
private:
    string NoPlat;
    string Merk;
    int TahunProduksi;
    string Warna;

public:
    Vehicle(string NoPlat, string Merk, int TahunProduksi, string Warna)
        : NoPlat(NoPlat), Merk(Merk), TahunProduksi(TahunProduksi), Warna(Warna) {}

    void setNoPlat(string NoPlat) {
        this->NoPlat = NoPlat;
    }

    string getNoPlat() const {
        return NoPlat;
    }

    void setMerk(string Merk) {
        this->Merk = Merk;
    }

    string getMerk() const {
        return Merk;
    }

    void setTahunProduksi(int TahunProduksi) {
        this->TahunProduksi = TahunProduksi;
    }

    int getTahunProduksi() const {
        return TahunProduksi;
    }

    void setWarna(string Warna) {
        this->Warna = Warna;
    }

    string getWarna() const {
        return Warna;
    }

    virtual void printDetails() const = 0;
};